src/main/java/org/gridsuite/modification/server/modifications/VscModification.java

```diff
@@ -51,10 +51,22 @@ public void check(Network network) throws NetworkModificationException {
                 || modificationInfos.getConverterStation2() == null) {
             throw new NetworkModificationException(MODIFY_BATTERY_ERROR, "Missing required attributes to modify the equipment");
         }
-        VscConverterStation converterStation1 = ModificationUtils.getInstance().getVscConverterStation(network, modificationInfos.getConverterStation1().getEquipmentId());
-        VscConverterStation converterStation2 = ModificationUtils.getInstance().getVscConverterStation(network, modificationInfos.getConverterStation2().getEquipmentId());
+        HvdcLine hvdcLine = ModificationUtils.getInstance().getHvdcLine(network, modificationInfos.getEquipmentId());
+
+        VscConverterStation converterStation1 = ModificationUtils.getInstance().getVscConverterStation(network, hvdcLine.getConverterStation1().getId());
+        VscConverterStation converterStation2 = ModificationUtils.getInstance().getVscConverterStation(network, hvdcLine.getConverterStation2().getId());
         checkConverterStation(modificationInfos.getConverterStation1(), converterStation1);
         checkConverterStation(modificationInfos.getConverterStation2(), converterStation2);
+
+        checkToModifyDroop();
+
+    }
+
+    private void checkToModifyDroop() {
```

Comment:
It's my opinion but I don't find the name appropriate.
Suggestion: checkDroopModification, isDropModificationAllowed...

Reply:
done with isDropModificationAllowed